hbase-migration-tools/bigtable-hbase-replication/README.md

```diff
@@ -0,0 +1,290 @@
+data. Near zero downtime migrations include the following steps:
+
+1. Install HBase to Cloud Bigtable library on the HBase master and region
+   servers
+2. Configure Cloud
+   Bigtable [authentication](https://cloud.google.com/bigtable/docs/authentication)
+3. Update hbase-site.xml with destination cloud bigtable project and instance
+4. Add a CBT replication peer in HBase. On HBase shell
+   execute `add_peer '2', ENDPOINT_CLASSNAME => 'com.google.cloud.bigtable.hbase.replication.HbaseToCloudBigtableReplicationEndpoint'`
+   . Use add_peer options to enable replication for select tables.
+5. Immediately disable the CBT replication peer, this allows WAL logs to
+   accumulate on HDFS. On HBase shell execute:  `disable_peer '2'`
+6. Check the replicated tables by executing `list_replicated_tables` and enable
+   table level replication by executing `enable_table_replication "table_name"`
+7. Copy the existing data
+   using [offline migration tooling](https://cloud.google.com/architecture/hadoop/hadoop-gcp-migration-data-hbase-to-bigtable)
+8. After all the data is copied (and verified), enable CBT replication peer. On
+   HBase shell execute `enable_peer '2'`
+9. Eventually, replication will catch up and the 2 databases will be in sync. On
+   HBase shell execute `status 'replciation'` to check status of replication for
+   peer ("2" in this example)
+10. Run validation steps to ensure compatibility and performance on CBT
+11. Once ready, switch over to CBT
+    1. Turn down applications writing to HBase
+    2. Wait for replication to catch up (This should be fast in the absence of
+       new writes)
+    3. Turn on the applications that write directly to CBT.
+12. Deprecate the HBase cluster
+
+## Prerequisites
+
+- HBase cluster is configured with the setting `hbase.replication` to `true` in
+  hbase-site.xml
+- Cloud Bigtable instance is created
+- Cloud Bigtable instance has all the tables with all column families
+- Cloud Bigtable authentication is configured
+
+## HBase configuration
+
+Have the below properties set in `hbase-site.xml` and add it to the HBase region
+server classpath.
+
+```
+<property> 
+    <name>google.bigtable.project.id</name>
+    <value>PROJECT_KEY</value>
+    <description>
+       Cloud Bigtable Project ID
+    </description>
+</property>
+<property>
+    <name>google.bigtable.instance.id</name>
+    <value>INSTANCE_KEY</value>
+    <description>
+       Cloud Bigtable Instance ID
+    </description>
+</property>
+```
+
+Please refer
+to [HBaseToCloudBigtableReplicationConfiguration](bigtable-hbase-replication-core/src/main/java/com/google/cloud/bigtable/hbase/replication/configuration/HBaseToCloudBigtableReplicationConfiguration.java)
+for other properties that can be set. Please set other properties for
+authenticating to Cloud Bigtable based on
+your preferred [authentication](https://cloud.google.com/bigtable/docs/authentication)
```

Comment:
is there a more prescriptive page for configuring authentication? For on-prem deployments, it is common for services to authenticate using service accounts, which we may require setting configurations such as `google.bigtable.auth.json.keyfile` in the hbase-site.xml

Reply:
Maybe @igorbernstein2 or @kolea2  will know?